core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -129,7 +132,7 @@ private[spark] object BlockManagerId {
   def apply(in: ObjectInput): BlockManagerId = {
     val obj = new BlockManagerId()
     obj.readExternal(in)
-    getCachedBlockManagerId(obj)
+    obj
```

Comment:
Why would we not cache this ?

Reply:
`BlockManagerId`  caching has been moved into `readExternal` because its deserialization at  `CompressedMapStatus` or `HighlyCompressedMapStatus` have changed from `BlockManagerId.apply()` to `BlockManagerId.readExternal()`

Reply:
I do not see caching in `BlockManagerId` [here](https://github.com/apache/spark/pull/31876/files#diff-eeeec5a404b590db51683bcc967313f6e2e7fbc4c721e2756c69b8976da22b5aR81). What am I missing ?